metrics/prometheus-backend/src/main/scala/com/softwaremill/sttp/prometheus/PrometheusBackend.scala

```diff
@@ -0,0 +1,56 @@
+package com.softwaremill.sttp.prometheus
+
+import java.util.concurrent.ConcurrentHashMap
+
+import com.softwaremill.sttp.{FollowRedirectsBackend, MonadError, Request, Response, SttpBackend}
+import io.prometheus.client.Histogram
+
+import scala.collection.mutable
+import scala.language.higherKinds
+import scala.collection.JavaConverters._
+
+class PrometheusBackend[R[_], S] private (delegate: SttpBackend[R, S],
+                                          requestToHistogramNameMapper: Option[Request[_, S] => String])
+    extends SttpBackend[R, S] {
+
+  import PrometheusBackend._
+
+  private[this] val histograms: mutable.Map[String, Histogram] = new ConcurrentHashMap[String, Histogram]().asScala
+
+  override def send[T](request: Request[T, S]): R[Response[T]] = {
+    val histogramName = getHistogramName(request)
```

Comment:
maybe also add a request counter and a requests-in-progress counter?

Reply:
registering single histogram gives: `sttp_request_latency_sum`, `sttp_request_latency_count`, `sttp_request_latency `, `sttp_request_latency_bucket`, so we have a counter :)

regarding in-progress, hm, not so simple, we would need two counters (increment only) for 1) started and 2) finished https://prometheus.io/docs/concepts/metric_types/

Reply:
a gauge? :)

Reply:
done